client.go

```diff
@@ -571,10 +571,13 @@ func (client *client) update(data *MetadataResponse) ([]string, error) {
 				toRetry[topic.Name] = true
 			}
 		}
-		var partitionCache [maxPartitionIndex][]int32
-		partitionCache[allPartitions] = client.setPartitionCache(topic.Name, allPartitions)
-		partitionCache[writablePartitions] = client.setPartitionCache(topic.Name, writablePartitions)
-		client.cachedPartitionsResults[topic.Name] = partitionCache
+
+		if topic.Err != ErrUnknownTopicOrPartition && topic.Err != ErrInvalidTopic {
```

Comment:
ah whoops, I'm dumb, sorry.

I believe we may want to similarly delete from `client.metadata` and guard that creation though, still trying to figure that out